CONTRIBUTING.md

```diff
@@ -0,0 +1,90 @@
+and each part of the functionality of the bridge will be in a seperate class.
+For example, the processing of Matrix events destined for Discord are handled
+inside the `MatrixEventProcessor` class.
+
+## Setting up
+
+(This section needs to be written in detail)
+
+* You will need to [setup the bridge](https://github.com/Half-Shot/matrix-appservice-discord/tree/develop#setup-the-bridge) similarly to how we describe,
+  but you should setup a homeserver locally on your development machine. We would recommend [Synapse](https://github.com/matrix-org/synapse#id11).
+
+## Writing an issue
+
+When writing an issue, please be as verbose as you can. Remember the issue is
+there to either document a feature request, or report a bug so it can be fixed.
+The issue board is NOT there to complain about a broken or missing feature.
+
+We leave it to the author's discretion to decide what to include rather than
+provide a template, but good items are:
+ * Shorter titles are better than long rambling ones, but please don't make it too vague.
+    * A good example is "Ability to bridge an existing matrix room into a discord channel"
+ * A brief description of the problem.
+    * If you are a user of another person's bridge, please can you let us know the name of the service provider.
+    * If you would like to keep the details private, please PM @Half-Shot:half-shot.uk discreetly.
+ * Relevant logging from Synapse/your homeserver AND the bridge (if applicable).
+    * The more verbose, the better but please don't include sensitive details like access tokens.
+ * A screenshot is always useful.
+ * Please mention which direction a failure is in, e.g. Matrix -> Discord, if applicable.
+
+We will assign each issue a tag, which will allow us to categorise the problems.
+
+While we realise some issues are more important than others, please do not "demand"
+for an issue to be fixed. Issues will be worked on in the timeframe that best fits
+the needs of the team.
+
+
+## PR Process
+
+We've tried our best to keep the PR process relatively simple:
+
+* Create a new branch based off the `develop` branch.
+    * This can be done with `git checkout develop` followed by `git checkout -b featurename`.
+* Create a PR on Github, making sure to give a brief discription of your changes and link to the issue it fixes, if any.
+    * If your change is not complete but you would like feedback, create it as a draft.
+* Ensure the linter and tests are not failing, as we will not accept code that breaks either.
+    * If your tests fail and are having trouble fixing them, you may push your changes and we will help you fix them.
+    * Github automatically pokes TravisCI to run both linting and tests.
+* Someone from the team will review your work and decide what to do with the PR.
+    * Usually we will have feedback for the PR and will submit more comments.
+    * We may decide to reject it, if a feature has little chance of fitting in with the project goals.
```

Comment:
`* Please don't take criticism personally. We try to keep the bridge high-quality!` or something like that?

Reply:
I think it's hopefully implied by the CoC statement